modules/utxo-lib/src/bitgo/zcash/ZcashPsbt.ts

```diff
@@ -22,6 +32,68 @@ export class ZcashPsbt extends UtxoPsbt<ZcashTransaction<bigint>> {
     );
   }
 
+  /**
+   * In version < 5 of Zcash transactions, the consensus branch ID is not serialized in the transaction
+   * whereas in version 5 it is. If the transaction is less than a version 5, set the consensus branch id
+   * in the global map in the psbt. If it is a version 5 transaction, throw an error if the consensus
+   * branch id is set in the psbt (because it should be on the transaction already).
+   * @param buffer Psbt buffer
+   * @param opts options
+   */
+  static fromBuffer(buffer: Buffer, opts: PsbtOpts): UtxoPsbt<UtxoTransaction<bigint>> {
+    const psbt = super.fromBuffer(buffer, opts) as ZcashPsbt;
+
+    // Read `consensusBranchId` from the global-map
+    let consensusBranchId: number | undefined = undefined;
+    psbt.data.globalMap.unknownKeyVals?.forEach(({ key, value }, i) => {
+      if (key.equals(CONSENSUS_BRANCH_ID_KEY)) {
+        consensusBranchId = value.readUint32BE();
+      }
+    });
+    switch (psbt.tx.version) {
+      case 4:
+      case ZcashTransaction.VERSION4_BRANCH_CANOPY:
+      case ZcashTransaction.VERSION4_BRANCH_NU5:
+        if (!consensusBranchId || !psbt.data.globalMap.unknownKeyVals) {
```

Comment:
Do we need the second part of this OR statement? I think if `psbt.data.globalMap.unknownKeyVals` is undefined, then `consensusBranchId` will always be undefined already.

Reply:
It could be that there are other values in the global map (in the future for musig this will be the case) and we do not find the consensus branch id in the map